zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -55,84 +68,189 @@ private ZeebeRocksDbFactory(
+    statistics.setStatsLevel(StatsLevel.ALL);
 
-    if (columnFamilyNames != null && !columnFamilyNames.isEmpty()) {
-      for (final byte[] name : columnFamilyNames) {
-        final ColumnFamilyDescriptor columnFamilyDescriptor =
-            new ColumnFamilyDescriptor(name, columnFamilyOptions);
-        columnFamilyDescriptors.add(columnFamilyDescriptor);
-      }
-    }
-    return columnFamilyDescriptors;
+    return new DBOptions()
+        .setErrorIfExists(false)
+        .setCreateIfMissing(true)
+        .setParanoidChecks(true)
+        // 1 flush, 1 compaction
+        .setMaxBackgroundJobs(2)
+        // we only use the default CF
+        .setCreateMissingColumnFamilies(false)
+        // may not be necessary when WAL is disabled, but nevertheless recommended to avoid
+        // many small SST files
+        .setAvoidFlushDuringRecovery(true)
+        // fsync is called asynchronously once we have at least 4Mb
+        .setBytesPerSync(4 * 1024 * 1024L)
+        // limit the size of the manifest (logs all operations), otherwise it will grow
+        // unbounded
+        .setMaxManifestFileSize(256 * 1024 * 1024L)
+        // speeds up opening the DB
+        .setSkipStatsUpdateOnDbOpen(true)
+        // keep 1 hour of logs - completely arbitrary. we should keep what we think would be
+        // a good balance between useful for performance and small for replication
+        .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
+        .setKeepLogFileNum(2)
+        // can be disabled when not profiling
+        .setStatsDumpPeriodSec(20)
+        .setStatistics(statistics);
   }
 
   /** @return Options which are used on all column families */
-  public ColumnFamilyOptions createColumnFamilyOptions() {
-    // start with some defaults
-    final var columnFamilyOptionProps = new Properties();
-    // look for cf_options.h to find available keys
-    // look for options_helper.cc to find available values
-    columnFamilyOptionProps.put("compaction_pri", "kOldestSmallestSeqFirst");
-
-    // apply custom options
-    columnFamilyOptionProps.putAll(userProvidedColumnFamilyOptions);
-
-    final var columnFamilyOptions =
-        ColumnFamilyOptions.getColumnFamilyOptionsFromProps(columnFamilyOptionProps);
-    if (columnFamilyOptions == null) {
-      throw new IllegalStateException(
-          String.format(
-              "Expected to create column family options for RocksDB, "
-                  + "but one or many values are undefined in the context of RocksDB "
-                  + "[Compiled ColumnFamilyOptions: %s; User-provided ColumnFamilyOptions: %s]. "
-                  + "See RocksDB's cf_options.h and options_helper.cc for available keys and values.",
-              columnFamilyOptionProps, userProvidedColumnFamilyOptions));
+  public ColumnFamilyOptions createColumnFamilyOptions(final List<AutoCloseable> closeables) {
+
+    final var hasUserOptions = !userProvidedColumnFamilyOptions.isEmpty();
```

Comment:
Above we don't use var, and here we do - I have a slight preference to not use var anymore (happy to discuss why somewhere else :sweat_smile:), but it's not a strong opinion, so here I'd just prefer we be at least consistent in the same file.